# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/spellchecker/TextServicesDocument.h
# Commit: 6af3e577108b
# Full Hash: 6af3e577108bbd68891f17a5b002fb88195f4ca3
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-07-13 21:42:32
# Regressor Bug: 1718924
# File Overlap Count: 2
# Description:
#   Bug 1718924 - part 18: Move some blocks which work with `TextServicesDocument::mOffsetTable` into new separated methods r=m_kato
#   
#   Depends on D119164
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119165
# ==============================================================================

diff -r 3b175989fe8c -r 6af3e577108b editor/spellchecker/TextServicesDocument.h
--- a/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 11:16:00 2021 +0000
+++ b/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 11:19:22 2021 +0000
@@ -94,6 +94,28 @@
      */
     void RemoveInvalidElements();
 
+    /**
+     * Called when non-collapsed selection will be deleted.
+     */
+    nsresult WillDeleteSelection();
+
+    /**
+     * Called when non-collapsed selection is deleteded.
+     */
+    OffsetEntry* DidDeleteSelection();
+
+    /**
+     * Called when aInsertedText is inserted.
+     */
+    MOZ_CAN_RUN_SCRIPT nsresult DidInsertText(dom::Selection* aSelection,
+                                              const nsAString& aInsertedString);
+
+    /**
+     * Called when selection range will be applied to the DOM Selection.
+     */
+    Result<EditorRawDOMRangeInTexts, nsresult> WillSetSelection(
+        uint32_t aOffsetInTextInBlock, uint32_t aLength);
+
     class Selection final {
      public:
       size_t StartIndex() const {
